src/weather_gen.cpp

```diff
@@ -23,6 +23,43 @@ constexpr double tau = 2 * PI;
 weather_generator::weather_generator() = default;
 int weather_generator::current_winddir = 1000;
 
+double weather_generator::get_weather_temperature( const tripoint &location, const time_point &t,
```

Comment:
This looks like it's largely copied from `get_weather`. What if someone changes the calculation in the other function, but not here? The two functions may generate different temperatures for the same input.

Either call the main function directly and get rid of this copy, or add another function that calculates the common stuff and that gets called from both functions (`get_weather` and this one).